.github/workflows/linter.yaml

```
@@ -5,6 +5,8 @@ on:
       - main
       - incubation
   pull_request:
+permissions:
+  checks: write
```

I think we can add all suggested and also enable `only-new-issues` on the job (so the 2nd one is relevant). WDYT?

```suggestion
permissions:
  contents: read
  pull-requests: read
  checks: write
```

---

just feel very strange from [this](https://github.com/golangci/golangci-lint-action#comments-and-annotations) 
like if we still need to set contents: read, but if we do not have it, how this GH could even work 
pull-requests might not need coz we do not have only-new-issues, 
anyway, use whats recommends from doc, we trust docs